Modify NXP's CSI driver to support both rt11xx and rt10xx

ngphibang opened the pull request:
This PR modify the NXP's CSI driver to support both rt11xx and rt10xx. This depends on [this PR](https://github.com/zephyrproject-rtos/zephyr/pull/71859)
It is split from https://github.com/zephyrproject-rtos/zephyr/pull/69810 as required by @loicpoulain to ease the review process.
It already got reviewed by @danieldegrasse.

The PR depends on https://github.com/zephyrproject-rtos/zephyr/pull/71859

ngphibang commented:
The CI is failed because of the dependency on [this PR](https://github.com/zephyrproject-rtos/zephyr/pull/71859)

danieldegrasse commented on drivers/video/video_mcux_csi.c:
```diff
@@ -132,9 +165,17 @@ static int video_mcux_csi_set_fmt(const struct device *dev,
+		return -ENOTSUP;
+	}
+	video_pix_fmt_convert(&format, false);
+	data->csi_config.dataBus = kCSI_DataBus24Bit;
```
Maybe instead of making this dependent on Kconfig we should move the `dataBus` configuration to devicetree? Not sure if all cameras are 8bit bus width, or if some would use more pins

ngphibang replied:
We can't make this configurable in devicetree. The databus is either 8 bits or 24 bits depending on the path of image data. If it passes by mipi csi2rx, data are implicitly converted to 32-bit format and 24 bits need to be set for it to work.

danieldegrasse commented on include/zephyr/drivers/video.h:
Move the change to this generic file into a separate commit from the one adding support for the CSI2RX to the CSI driver

ngphibang replied:
Okay. Done

ngphibang committed:
include: video: Add XYUV32 pixel format

Add XYUV32 pixel format

Signed-off-by: Phi Bang Nguyen <phibang.nguyen@nxp.com>

ngphibang committed:
drivers: video: csi: Add support for i.MX RT11XX

On i.MX RT11XX which has MIPI CSI-2 Rx, image data from the camera sensor
after passing through the camera pipeline (MIPI CSI-2 Rx --> Video Mux -->
CSI) will be implicitly converted to a 32-bits pixel formats. For example,
an input in RGB565 / YUYV (2-bytes format) will become an XRGB32 / XYUV32
(4-bytes format), respectively. Make changes to support this.

Signed-off-by: Phi Bang Nguyen <phibang.nguyen@nxp.com>

danieldegrasse commented:
@ngphibang please take a look at CI failures

ngphibang commented:
@danieldegrasse Yes, CI failures are just because of dependencies on other PRs. We can wait for them to be merged first then I will update this.

ngphibang commented:
Rebased because https://github.com/zephyrproject-rtos/zephyr/pull/72623 is now merged.

dleach02 commented:
You have a Kconfig compliance check error to address

ngphibang commented:
@dleach02 As being said in my previous comment to Daniel, the undefined Kconfig error `CONFIG_VIDEO_MCUX_MIPI_CSI2RX` is just because of the dependency on https://github.com/zephyrproject-rtos/zephyr/pull/71859. Once this one is merged, it will be ok.

ngphibang commented:
https://github.com/zephyrproject-rtos/zephyr/pull/71859 is now merged.
@loicpoulain @danieldegrasse : Could you approve this again ?

The CI failures are not due to my changes, I think.

```
HEAD is now at 75d0880 tests: Update mps2 board name
ERROR: update failed for projects: psa-arch-tests, sof, tflite-micro, thrift, hal_espressif, hal_nxp
Error: Error: failed to run script step: command terminated with non-zero exit code: Error executing in Docker Container: 1
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.


Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --prune --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*
  Error: error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
  Error: error: 7507 bytes of body are still expected
  fetch-pack: unexpected disconnect while reading sideband packet
  Error: fatal: early EOF
  Error: fatal: fetch-pack: invalid index-pack output
  The process '/usr/bin/git' failed with exit code 128
  Waiting 15 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --prune --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*
  Error: error: RPC failed; curl 92 HTTP/2 stream 0 was not closed cleanly: CANCEL (err 8)
  Error: error: 574 bytes of body are still expected
  fetch-pack: unexpected disconnect while reading sideband packet
  Error: fatal: early EOF
  Error: fatal: fetch-pack: invalid index-pack output
  The process '/usr/bin/git' failed with exit code 128
  Waiting 12 seconds before trying again
  /usr/bin/git -c protocol.version=2 fetch --prune --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/* +refs/tags/*:refs/tags/*
  Error: error: 6868 bytes of body are still expected
  fetch-pack: unexpected disconnect while reading sideband packet
  Error: fatal: early EOF
  Error: fatal: fetch-pack: invalid index-pack output
  Error: The process '/usr/bin/git' failed with exit code 128
```

danieldegrasse reviewed (approved):
Will rekick CI once things are working again, the failures do not seem to be related to this PR
